### ISSUE TITLE
Training KIE model on custom dataset getting very low F1-score

### ISSUE THREAD

**Issue by adesgautam** [open-mmlab/mmocr#363]
Hi,
I have been trying to train the KIE SDMGR model on my custom dataset prepared by sending the documents to tesseract and then labelling all of the text to some classes. 
I have been using whole lines without spaces as text and bboxes returned from tesseract. 
The dataset is prepared just like wildreciept dataset.
But the problem is when i am training it on the data, the f1-score is very low, low as 0.01XX. 
Can anyone help me in this ? 
Or any tips to improve the model, or if there is an issue with the dataset created ?

Edit: Also can someone also tell why are there no spaces in dictionary for the SDMGR model in the dataset ? Can i add it ?

**Comment by amitbcp** [comment]
@adesgautam how many labels do you in your custom dataset ?

In the training config script, you will see some class ids are ignored by for evaluation. You might want to make that list empty and try

**Comment by adesgautam** [comment]
I have a total of 26 classes.

evaluation = dict(
    interval=1,
    metric='macro_f1',
    metric_options=dict(
        macro_f1=dict(
            ignores=[0, 1, 3])))

Setting ignores as empty you mean ?

**Comment by amitbcp** [comment]
evaluation = dict(
interval=1,
metric='macro_f1',
metric_options=dict(
macro_f1=dict(
ignores=[])))


Setting ignore to an empty list. 

Also, it seems you are using the whole sentence text as your input for key/values. The model uses a BiLSTM for textual embedding, so you might want to change the length of that.

In my case what I did was, I broke the sentences too words and assigned each word the same label rather than taking the whole sentence together. Later in inference I just did post processing to stitch it back. It worked for me. You can give that a shot

**Comment by adesgautam** [comment]
Okay thanks for that, i'll check it out.
Also, have you also added spaces in the dictionary ? Or is it not required and going with word will workout fine ?

**Comment by cuhk-hbsun** [comment]
> Okay thanks for that, i'll check it out.
> Also, have you also added spaces in the dictionary ? Or is it not required and going with word will workout fine ?

nope, you can just remove the space from your ground-truth "text". It is not necessary.

**Comment by adesgautam** [comment]
Thanks for the help. 
Actually the bounding boxes were not correct. 
After fixing it the F1-score is reaching 0.9+ and predictions are good.

**Comment by amitbcp** [comment]
ohh great @adesgautam ! you can close the issue then

**Comment by manjrekarom** [comment]
@adesgautam is it possible for you to link your training script for KIE here?

**Comment by adesgautam** [comment]
It's already there in tools/train.py, just use the kie config file
